modules/ximgproc/src/fast_line_detector.cpp

```diff
@@ -544,7 +545,8 @@ void FastLineDetectorImpl::lineDetection(const Mat& src, std::vector<SEGMENT>& s
     std::vector<Point2i> points;
     std::vector<SEGMENT> segments, segments_tmp;
     Mat canny;
-    Canny(src, canny, canny_th1, canny_th2, canny_aperture_size);
+    if (canny_aperture_size == 0) canny = src;
+    else Canny(src, canny, canny_th1, canny_th2, canny_aperture_size);
```

Comment:
Consider using dedicated lines for if/else bodies.
It is necessary to see real picture from "code coverage" line-based reports ([example of weekly report](http://pullrequest.opencv.org/buildbot/export/opencv_releases/master-contrib_coverage-lin64-debug/20201017-080737_10545/coverage_html/index.html)).